### PR TITLE
add-Num#bigDecimalValue

### PR DESCRIPTION
**Pull request by nimo23**
Fixes #.

Changes proposed in this pull request:
- added `Num.bigDecimalValue()` to convert the `Num` to a `BigDecimal`
    - For `DecimalNum` it returns its `delegate`
    - For `DoubleNum` it converts by `BigDecimal.valueOf(delegate)` (if delegate is `NaN` or `Infinity`, then it simply returns `null`)
    - For `NaN` it simply returns `null`.

- [x] added an entry with related ticket number(s) to the unreleased section of `CHANGES.md` 

_Actually, it's not easy to get the `BigDecimal`-representation of a `Num`. We have `intValue()`, `longValue()`, `doubleValue()`, etc. so it also makes sense to add `bigDecimalValue()`._

### CODE REVIEW

**Comment by TheCookieLab** [comment]
> Would it be better to overwrite `bigDecimalValue()` in its subclasses instead of providing a default method?
> 
> For example, in `DecimalNum`:
> 
> ```
>  @Override
>  public BigDecimal bigDecimalValue() {
>         return delegate;
> }
> ```
> 
> With this, no **if** and **cast** is needed..

Yes, that would be ideal.

File: CHANGELOG.md [hunk 1]
```diff
@@ -132,6 +132,7 @@ Changelog for `ta4j`, roughly following [keepachangelog.com](http://keepachangel
 - added "TradeExecutionModel" to modify trade execution during backtesting
 - added **NumIndicator** to calculate any `Num`-value for a `Bar`
 - added **RunningTotalIndicator** to calculate a cumulative sum for a period.
+- added `Num.bigDecimalValue(DoubleNum)` to convert Num to a BigDecimal
```
Comment (TheCookieLab) [inline CHANGELOG.md]: This should be moved up to the 0.17 section

File: ta4j-core/src/main/java/org/ta4j/core/num/Num.java [hunk 2]
```diff
@@ -354,6 +345,31 @@ default float floatValue() {
...
+        return getDelegate().doubleValue();
+    }
+
+    /**
+     * Converts this {@code Num} to a {@code BigDecimal}.
+     *
+     * @return this {@code Num} converted to a {@code BigDecimal}
+     */
+    default BigDecimal bigDecimalValue() {
+        Number num = getDelegate();
```
Comment (TheCookieLab) [inline ta4j-core/src/main/java/org/ta4j/core/num/Num.java]: Can we add unit tests to **DoubleNumTest** and **DecimalNumTest** to exercise this branching logic, including the doubleNum edge cases?

**Review by TheCookieLab — APPROVED** [review]
